src/pyformance/meters/gauge.py

```diff
@@ -0,0 +1,60 @@
+from typing import Callable, Union
+
+
+class Gauge[T: Union[int, float]]:
```

Comment:
The generic seems like a good way to type it, but it does cause a lot of complexity down the road. I expect it would change little if I would type as the union instead of bounded by the union. 

Wdyt?

Reply:
I like the type var, and I think its impact is quite OK. especially if you'd write it as `int | float` it's quite compact. But I'm not opposed to your suggestion.

FYI, Python 3.13 addresses this problem with defaults for type vars. e.g. you could declare `class Gauge[T: int | float = int | float]:` and use it as `Gauge` to mean `Gauge[int | float]`.